test/Microsoft.TestPlatform.AcceptanceTests/CodeCoverageTests.cs

```diff
@@ -28,60 +55,156 @@ public CodeCoverageTests()
+                RunSettingsPath = string.Empty,
+                RunSettingsType = TestParameters.SettingsType.Default,
+                ExpectedPassedTests = 1,
+                ExpectedSkippedTests = 1,
+                ExpectedFailedTests = 1
+            };
+
+            this.CollectCodeCoverage(runnerInfo, parameters);
         }
 
         [TestMethod]
         [NetFullTargetFrameworkDataSource(useDesktopRunner: false)]
         [NetCoreTargetFrameworkDataSource(useDesktopRunner: false)]
         public void CollectCodeCoverageX64WithRunSettings(RunnerInfo runnerInfo)
         {
-            this.CollectCodeCoverage(runnerInfo, "x64", withRunsettings: true);
+            var parameters = new TestParameters()
+            {
+                AssemblyName = "SimpleTestProject.dll",
+                TargetPlatform = "x64",
+                RunSettingsPath = string.Empty,
+                RunSettingsType = TestParameters.SettingsType.Default,
+                ExpectedPassedTests = 1,
+                ExpectedSkippedTests = 1,
+                ExpectedFailedTests = 1
+            };
+
+            this.CollectCodeCoverage(runnerInfo, parameters);
+        }
+
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource(useDesktopRunner: false)]
+        [NetCoreTargetFrameworkDataSource(useDesktopRunner: false)]
+        public void CodeCoverageShouldAvoidExclusionsX86(RunnerInfo runnerInfo)
+        {
+            var parameters = new TestParameters()
+            {
+                AssemblyName = "CodeCoverageTest.dll",
+                TargetPlatform = "x86",
+                RunSettingsPath = Path.Combine(
+                    IntegrationTestEnvironment.TestPlatformRootDirectory,
+                    @"scripts\vstest-codecoverage2.runsettings"),
+                RunSettingsType = TestParameters.SettingsType.Custom,
+                ExpectedPassedTests = 2,
+                ExpectedSkippedTests = 0,
+                ExpectedFailedTests = 0,
+                CheckSkippedMethods = true
+            };
+
+            this.CollectCodeCoverage(runnerInfo, parameters);
+        }
+
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource(useDesktopRunner: false)]
+        [NetCoreTargetFrameworkDataSource(useDesktopRunner: false)]
+        public void CodeCoverageShouldAvoidExclusionsX64(RunnerInfo runnerInfo)
+        {
+            var parameters = new TestParameters()
+            {
+                AssemblyName = "CodeCoverageTest.dll",
+                TargetPlatform = "x64",
+                RunSettingsPath = Path.Combine(
+                    IntegrationTestEnvironment.TestPlatformRootDirectory,
+                    @"scripts\vstest-codecoverage2.runsettings"),
+                RunSettingsType = TestParameters.SettingsType.Custom,
+                ExpectedPassedTests = 2,
+                ExpectedSkippedTests = 0,
+                ExpectedFailedTests = 0,
+                CheckSkippedMethods = true
```

Comment:
I think it would be easier to have inside TestParameters:

ExpectedSkippedMethods - list of strings
ExpectedCoveredMethods - list of strings

Reply:
This is needed in order to validate the test results summary as seen in method `ValidateSummaryStatus`. By reusing `CollectCodeCoverage` method, that among others does this validation step (other tests other than the ones written by me use this as well), it's mandatory to provide the number of passed/skipped/failed tests like this.

Actual validation of test names is done when reading the code coverage report and making sure one test is covered and the other is skipped.